cpp/include/cudf/detail/iterator.cuh

```diff
@@ -134,6 +178,37 @@ auto make_null_replacement_iterator(column_device_view const& column,
       null_replaced_value_accessor<Element>{column, null_replacement});
 }
 
+/**
+ * @brief Constructs a pair iterator over a column's values and its validity.
+ *
+ * Dereferencing the returned iterator returns a `thrust::pair<Element, bool>`.
+ * 
+ * If an element at position `i` is valid (or `has_nulls == false`), then for `p = *(iter + i)`, `p.first` contains 
+ * the value of the element at `i` and `p.second == true`.
+ *
+ * Else, if the element at `i` is null, then the value of `p.first` is undefined and `p.second == false`. 
+ * `pair(column[i], validity)`.
+ * `validity` is `true` if `has_nulls=false`.
+ * `validity` is validity of the element at `i` if `has_nulls=true` and the
+ * column is nullable.
+ *
+ * @throws `cudf::logic_error` if the column is nullable.
+ * @throws `cudf::logic_error` if column datatype and Element type mismatch.
+ *
+ * @tparam Element The type of elements in the column
+ * @tparam has_nulls boolean indicating to treat the column is nullable
+ * @param column The column to iterate
+ * @return auto Iterator that returns valid column elements, and validity of the
+ * element in a pair
+ */
+template <typename Element, bool has_nulls=false> 
+auto make_pair_iterator(column_device_view const& column)
```

Comment:
Would it also make sense to have a function that takes a plain column_view?

Reply:
I thought about it. But `cudf::column_device_view::create()` returns a unique pointer which will be deleted (and frees children as well) when the `make_pair_iterator` returns.

Reply:
@nvdbaranec